pkg/credentials/credential.go

```diff
@@ -31,12 +33,26 @@ func credentialFromDockerAuthConfig(authCfg types.AuthConfig) (Credential, error
 		return Credential{}, err
 	}
 
+	tool, ctx, err := toolNameAndCtxFromAddress(strings.TrimPrefix(authCfg.ServerAddress, "https://"))
+	if err != nil {
+		return Credential{}, err
+	}
+
 	return Credential{
-		ToolName: strings.TrimPrefix(authCfg.ServerAddress, "https://"),
+		Context:  ctx,
+		ToolName: tool,
 		Env:      env,
 	}, nil
 }
 
-func (c Credential) String() string {
-	return c.ToolName
+func toolNameWithCtx(toolName, credCtx string) string {
+	return toolName + "///" + credCtx
```

Comment:
Open for suggestions for something better than this. I need a way of combining toolName and credCtx such that it still creates a valid URL.